nomad/csi_endpoint.go

```diff
@@ -368,10 +368,13 @@ func (v *CSIVolume) Claim(args *structs.CSIVolumeClaimRequest, reply *structs.CS
 		return fmt.Errorf("missing volume ID")
 	}
 
+	isNewClaim := args.Claim != structs.CSIVolumeClaimGC &&
+		args.State == structs.CSIVolumeClaimStateTaken
+
 	// COMPAT(1.0): the NodeID field was added after 0.11.0 and so we
 	// need to ensure it's been populated during upgrades from 0.11.0
 	// to later patch versions. Remove this block in 1.0
-	if args.Claim != structs.CSIVolumeClaimRelease && args.NodeID == "" {
+	if isNewClaim && args.NodeID == "" {
```

Comment:
nit: un-touched in this PR but GH is displaying it, if a watchset is never used nil can be used instead https://github.com/hashicorp/nomad/pull/9323/files#diff-6a3c30b17bb256ab9e3d5bab1620c3b9afee29aaa06371eae6cae0951a2fb12aL376

Reply:
Cool, thanks! I have another PR I'm working on that'll touch that area of code, so I'll fix it in that one.